File: Army(java)/Warriors/Demon.java
package Warriors;

import EnumID.*;
import Basic.*;
import State.*;

public class Demon extends Soldier {
    private WarriorID warrior;

    public Demon(String name, int hp, int dmg) {
        super(name, hp, dmg);
        this.warrior = WarriorID.DEMON;
    }

    public WarriorID getWarriorID() {
        return warrior;
    }
}

File: Army(c++)/Warriors/Demon.cpp
#include "Demon.hpp"

Demon::Demon(std::string name, int hp, int dmg) 
    : Soldier(name, hp, dmg), warrior(WarriorID::DEMON) {}

Demon::~Demon(){}

WarriorID Demon::getWarriorID() const {
    return this->warrior;
}
